src/stack/stack.ts

```diff
@@ -0,0 +1,43 @@
+interface IStack<T> {
+  push: (element: T) => number,
+  pop: () => T | undefined,
+  peek: () => T | undefined,
+  isFull: () => boolean,
+  isEmpty: () => boolean,
+}
+
+export class Stack<T> implements IStack<T> {
+  private elements: T[];
+  private maxSize: number;
+
+  constructor(maxSize = Infinity) {
+    this.elements = [];
+    this.maxSize = maxSize;
+  }
+
+  push(element: T) {
+    if (this.isFull()) {
+      throw new Error('Error [push]: stack is full');
+    }
+
+    this.elements.push(element);
+
+    return this.elements.length;
+  }
+
+  pop() {
+    return this.elements.pop();
+  }
+
+  peek() {
+    return this.elements[this.elements.length - 1];
+  }
+
+  isFull() {
+    return this.elements.length === this.maxSize;
+  }
+
+  isEmpty() {
+    return this.elements.length === 0;
```

Comment:
Is this a performance optimization or we can just use `!return this.elements.length`?

Reply:
It's just a habit:) I find it more readable for me personally.